### DISCUSSION TITLE
RustDesk self hosted and the key

### DISCUSSION THREAD

**Discussion by Weaselfox** [rustdesk/rustdesk#9631] [Q&A]
Hi guys,

I just installed the RustDesk self hosted version (I think its v. 1.3.1) in a proxmox container with the provided install script (not via docker). Of course I want the system to be secure I was wondering if the key now is mandatory or not or if I somehow have to start the system with the k parameter (I'm not a linux pro :-) ). So I tested the "out of the box system" (means no changes so far) with and without keys and I could see the following behaviour:
If I start a remote session from computer A to remotely control the computer B and I don't provide any key I get the error key mismatch. So far so good. If I provide the key to computer A only, so to the computer who initiates the remote connection and the computer B has no key or even a wrong key set, the connection is also possible. In principle this is exactly what I am looking for, so only those guys who have the key can initiate remote sessions but clients can get remote support without key. The only thing I am wondering now is: Is this connection now encrypted or not? Or only the A - Server connection is encrypted and the B - Server connection isn't?

Thanks in advance for any answers...

Regards...

**Comment by rustdesk** [comment]
#763, go to [discord](https://discord.gg/nDceKgxnkV) please, please search there first, many guys have asked similar question because you are not the most special though you always think you are. Usually nobody will help you with self-hosting here.
